sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/engine/RealSyncEngine.kt

```diff
@@ -37,36 +41,61 @@ import timber.log.Timber
 
 @ContributesBinding(scope = AppScope::class)
 class RealSyncEngine @Inject constructor(
-    private val syncRepository: SyncRepository,
-    private val syncApi: SyncApiClient,
+    private val syncApiClient: SyncApiClient,
     private val syncScheduler: SyncScheduler,
     private val syncStateRepository: SyncStateRepository,
     private val plugins: PluginPoint<SyncablePlugin>,
 ) : SyncEngine {
 
-    override fun syncNow() {
-        Timber.d("Sync: petition to sync now")
+    override fun syncNow(trigger: SyncTrigger) {
+        Timber.d("Sync: petition to sync now trigger: $trigger")
+        when (trigger) {
+            BACKGROUND_SYNC -> scheduleSync()
+            APP_OPEN -> performSync()
+            FEATURE_READ -> performSync()
```

Comment:
the difference between the two methods is that one checks for `lastModified` date and the other one doesn’t. `performSync` will only send entries that have changed since the last sync, `sendAllData` sends everything.